main.go

```diff
@@ -79,15 +81,19 @@ func run(c *cli.Context) error {
 }
 
 func (client *Client) setLogLevel(level string) error {
-	if level != "info" && level != "debug" && level != "error" {
-		return fmt.Errorf("invalid log level specified (%s)", level)
-	}
 	response, err := client.httpc.Post("http://unix/v1/loglevel",
 		"application/x-www-form-urlencoded",
 		strings.NewReader(fmt.Sprintf("level=%v", level)))
 	if err != nil {
 		return err
 	}
+	if response.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+		return errors.New(strings.TrimRight(string(body), "\n"))
+	}
```

Comment:
So the change here is to not limit to 3 levels, but allow any input and see if setting that errors out ? 
I'm unfamiliar with context here, but do we want to print entire response body ? Depending on request couldn't it be a bunch of random junk.

Reply:
We have a CLI (this) and a logserver (relevant code here https://github.com/rancher/rancher/blob/master/pkg/logserver/server.go#L57), first we checked the loglevel in the CLI but then we want to add TRACE as a new loglevel and I saw that the server (see linked code) actually already does the validation but we don't check that (`err` from http client does not have anything to do with status code). Let me know if this clears things up. I can add more context if needed.

Reply:
Ahh, makes more sense. I suppose if we get a random http error you'd want that printed here anyway.